src/adaptors/http2/http2_adaptor.c

```diff
@@ -359,11 +364,25 @@ static void free_http2_stream_data(qdr_http2_stream_data_t *stream_data, bool on
         qd_message_free(stream_data->message);
     }
 
+    //
+    // If the client/server closed the connection abruptly, we need to release the stream_data->curr_stream_data and
+    // stream_data->next_stream_data.
+    // This final decref of the delivery is going to free the associated message but before this message can be freed
+    // all stream data (body data) objects need to be freed. We do this here.
+    //
     if (stream_data->in_dlv && !stream_data->in_dlv_decrefed) {
+    	if (stream_data->curr_stream_data)
+    		qd_message_stream_data_release(stream_data->curr_stream_data);
+    	if (stream_data->next_stream_data)
+    		qd_message_stream_data_release(stream_data->next_stream_data);
         qdr_delivery_decref(http2_adaptor->core, stream_data->in_dlv, "HTTP2 adaptor in_dlv - free_http2_stream_data");
     }
 
     if (stream_data->out_dlv && !stream_data->out_dlv_decrefed) {
+    	if (stream_data->curr_stream_data)
+    		qd_message_stream_data_release(stream_data->curr_stream_data);
+    	if (stream_data->next_stream_data)
+    		qd_message_stream_data_release(stream_data->next_stream_data);
```

Comment:
same Q here (line 377)?

Reply:
Same answer as for the previous question